### PR TITLE
Added simple auth provider to the config api. Removed root level clientConfig

### PR DESCRIPTION
**Pull request by valentinyanakiev**

### CODE REVIEW

File: src/utils/config/client/aad-config/simple-auth.provider.config.entity.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Field, ObjectType } from '@nestjs/graphql';
```
Comment (techsmyth) [inline src/utils/config/client/aad-config/simple-auth.provider.config.entity.ts]: it is in the aad-config directory?

**Review by techsmyth — CHANGES_REQUESTED** [review]
Looks good - just the directory structure needs an update: 
- client ==> authentication-providers
- move the simpleAuthConfig out of an aad folder

**Review by atstojanov — COMMENTED** [review]
We need better structure, like:
```javascript
{
  "authenticationProviders": [
    {
      "name": "Simple",
      "label": "Log in",
      "icon": "",
      "enabled": true,
      "config": {}
    },
    {
      "name": "Microsoft",
      "icon": "",
      "label": "Log in with Microsft",
      "enabled": true,
      "config": {
        // microsoft specific configuration
      }
    },
    {
      "name": "Google",
      "icon": "",
      "label": "Log in with Google",
      "enabled": false,
      "config": {
        // google specific configuration
      }
    }
  ]
}
```

File: src/utils/config/client/aad-config/simple-auth.provider.config.entity.ts (hunk 1 above)
Comment (valentinyanakiev) [reply src/utils/config/client/aad-config/simple-auth.provider.config.entity.ts]: Good catch, fixed in 918ea36f025beb6911189a9fd093e34ebbc50b33

**Comment by valentinyanakiev** [comment]
> We need better structure, like:
> 
> ```js
> {
>   "authenticationProviders": [
>     {
>       "name": "Simple",
>       "label": "Log in",
>       "icon": "",
>       "enabled": true,
>       "config": {
>         // simple authentication specific configuration
>       }
>     },
>     {
>       "name": "Microsoft",
>       "icon": "",
>       "label": "Log in with Microsft",
>       "enabled": true,
>       "config": {
>         // microsoft specific configuration
>       }
>     },
>     {
>       "name": "Google",
>       "icon": "",
>       "label": "Log in with Google",
>       "enabled": false,
>       "config": {
>         // google specific configuration
>       }
>     }
>   ]
> }
> ```
> 
> Having plural in the property name suggests for `collection` not an object.

Addressed in 75308b5d27397c830822f92405e54007ddc82553

**Comment by atstoyanov** [comment]
`authEnabled` should be part of the whole config, not the AADConfig.

**Review by techsmyth — APPROVED** [review]

File: src/utils/config/authentication-providers/authentication.provider.config.entity.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  enabled?: boolean;
+
+  @Field(() => ConfigUnion, {
+    nullable: false,
+    description: 'Configuration of the authenticaiton provider',
+  })
+  config?: IAadConfig | ISimpleAuthProviderConfig;
+}
+
+export const ConfigUnion = createUnionType({
```
Comment (techsmyth) [inline src/utils/config/authentication-providers/authentication.provider.config.entity.ts]: This is AuthConfigUnion, not generic ConfigUnion

File: .env.simple.auth.provider [hunk 3]
```diff
@@ -0,0 +1,2 @@
+AUTH_SIMPLE_AUTH_ISSUER=localhost:3002
```
Comment (techsmyth) [inline .env.simple.auth.provider]: Is this file needed given there is a .env.simple.auth.provider.default?

**Comment by techsmyth** [comment]
update config to be 
authentication: {
    enabled: true,
   providers: [



  ]
}

File: .env.simple.auth.provider (hunk 3 above)
Comment (valentinyanakiev) [reply .env.simple.auth.provider]: Rename gone wrong - fixed it...

File: src/utils/config/authentication-providers/authentication.provider.config.entity.ts (hunk 2 above)
Comment (valentinyanakiev) [reply src/utils/config/authentication-providers/authentication.provider.config.entity.ts]: Fixed.

File: src/utils/config/authentication/providers/authentication.provider.config.entity.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  @Field(() => AuthenticationConfigUnion, {
+    nullable: false,
+    description: 'Configuration of the authenticaiton provider',
+  })
+  config?: IAadConfig | ISimpleAuthProviderConfig;
+}
+
+export const AuthenticationConfigUnion = createUnionType({
+  name: 'ConfigUnion',
```
Comment (techsmyth) [inline src/utils/config/authentication/providers/authentication.provider.config.entity.ts]: This needs to be AuthenticationProviderConfigUnion

File: src/utils/config/authentication/providers/authentication.provider.config.entity.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  types: () => [AadConfig, SimpleAuthProviderConfig],
```
Comment (techsmyth) [inline src/utils/config/authentication/providers/authentication.provider.config.entity.ts]: This should be "AadAuthProviderConfig" at least at the graphql api level so that that is consistent. Internal names etc we can / should pick up later.

**Review by techsmyth — APPROVED** [review]
Two naming updates still needed imho - with those in it is ready to go...

File: src/utils/config/authentication/providers/authentication.provider.config.entity.ts (hunk 4 above)
Comment (valentinyanakiev) [reply src/utils/config/authentication/providers/authentication.provider.config.entity.ts]: Fixed in 222a2c77eda25806d4848fc31e19b3d6846c8aee

File: src/utils/config/authentication/providers/authentication.provider.config.entity.ts (hunk 5 above)
Comment (valentinyanakiev) [reply src/utils/config/authentication/providers/authentication.provider.config.entity.ts]: Fixed in 222a2c77eda25806d4848fc31e19b3d6846c8aee

**Review by techsmyth — APPROVED** [review]
